service/hproxy/hproxy.go

```diff
@@ -566,6 +622,28 @@ func (s *Server) handleProxyRequest(w http.ResponseWriter, r *http.Request) {
 //	}
 // }
 
+func (s *Server) filterRequest(r *http.Request) error {
+	// copy and reset body
+	data, err := io.ReadAll(r.Body)
+	if err != nil {
+		maxBytesError := &http.MaxBytesError{}
+		if errors.As(err, &maxBytesError) {
```

Comment:
I think we can do this:
```suggestion
		var maxBytesError http.MaxBytesError
		if errors.As(err, &maxBytesError) {
```